src/mito2/src/schedule/scheduler.rs

```diff
@@ -16,16 +16,58 @@ use std::future::Future;
 use std::pin::Pin;
 use std::sync::atomic::{AtomicU8, Ordering};
 use std::sync::{Arc, RwLock};
+use std::time::Duration;
 
 use common_telemetry::warn;
+use futures::StreamExt;
 use snafu::{ensure, OptionExt, ResultExt};
 use tokio::sync::Mutex;
 use tokio::task::JoinHandle;
 use tokio_util::sync::CancellationToken;
 
 use crate::error::{InvalidSchedulerStateSnafu, InvalidSenderSnafu, Result, StopSchedulerSnafu};
+use crate::metrics::SCHEDULER_TASK_ELAPSED;
+use crate::schedule::priority;
+
+/// Priority of the job.
+pub enum Priority {
+    /// High priority job.
+    High,
+    /// Low priority job with an optional deadline.
+    Low(Option<Duration>),
+}
+
+pub struct Job {
+    /// Job type.
+    r#type: &'static str,
+    /// Job priority.
+    priority: Priority,
```

Comment:
I think we can provide a method that send by priority in the channel side.

Reply:
> I think we can provide a method that send by priority in the channel side.

Maybe add a priority argument to `Scheduler::schedule`:https://github.com/GreptimeTeam/greptimedb/blob/f3fc31708a3a213dbc12d59a42b4860d5f658532/src/mito2/src/schedule/scheduler.rs#L81